### ISSUE TITLE
Reference to variable "async" lost

### ISSUE THREAD

**Issue by ilpon** [fabiosantoscode/terser#87]
**Bug report or Feature request?**

bug

**Version (complete output of `terser -V`)**

```
$ node -v
v8.11.3
$ ./node_modules/.bin/terser -V
terser 3.8.0
```

**`terser` input**

```
$ cat async_variable.js 
const async = [ 'Hello World!' ];
console.log(async[0]);
```

**Expected result**

```
$ node async_variable.js 
Hello World!
```

**Actual result**

```
$ ./node_modules/.bin/terser async_variable.js | node
[ 0 ]
```

**`terser` output**

```
$ ./node_modules/.bin/terser async_variable.js 
const async=["Hello World!"];console.log([0]);
```

**Comment by kzc** [comment]
Use parentheses as a temporary workaround:
```
(async)[0]
```

Fix:

```patch
--- a/lib/parse.js
+++ b/lib/parse.js
@@ -2064,7 +2064,11 @@ function parse($TEXT, options) {
             return new_(allow_calls);
         }
         var start = S.token;
-        var async = is("name", "async") && peek().type != "arrow" && as_atom_node();
+        var peeked;
+        var async = is("name", "async")
+            && (peeked = peek()).value != "["
+            && peeked.type != "arrow"
+            && as_atom_node();
         if (is("punc")) {
             switch (S.token.value) {
               case "(":
```

@fabiosantoscode I couldn't resist. Feel free to commit the patch along with this test case:
```js
function async(async) {
    console.log(async[0], async.prop);
}
async({ 0: 1, prop: 2 });
```

**Comment by fabiosantoscode** [comment]
Thanks @kzc. Fixed as of v3.8.1
